karapace/kafka_rest_apis/consumer_manager.py

```diff
@@ -509,16 +509,30 @@ async def fetch(self, internal_name: Tuple[str, str], content_type: str, formats
                 for msg in poll_data[tp]:
                     try:
                         key = await self.deserialize(msg.key, request_format) if msg.key else None
-                    except (UnpackError, InvalidMessageHeader, InvalidPayload) as e:
-                        KarapaceBase.internal_error(
+                    except (
+                        UnpackError,
+                        InvalidMessageHeader,
+                        InvalidPayload,
+                        json.decoder.JSONDecodeError,
+                        UnicodeDecodeError,
```

Comment:
Handle these in the serde function and raise e.g. `DeserializationError` from there.

Reply:
👍🏼